File: 561. Array Partition I/solution.go
func arrayPairSum(nums []int) int {
	res := 0
	buckets := [20001]int{}

	for _, v := range nums {
		buckets[v+10000]++
	}

	odd := true
	for i := 0; i < len(buckets); i++ {
		for buckets[i] > 0 {
			if odd {
				res += (i - 10000)
			}

			odd = !odd
			buckets[i]--
		}
	}

	return res
}

File: 561. Array Partition I/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var arrayPairSum = function(nums) {
  return nums
    .sort((a, b) => a - b)
    .reduce((sum, cur, i) => (i % 2 === 0 ? (sum += cur) : sum), 0);
};
